### PR TITLE
Modified Create Options to pass config options to execution Provider

### PR DESCRIPTION
**Pull request by sfatimar**
### Description
Create Passes in a single buffer ProviderOptions and SessionOptions Config to the 
The Factory Creator which stores the reference to config and when createprovider is invoked
same reference is used to get config options

### Motivation and Context
<!-- - Why is this change required? What problem does it solve?
This issue isfixing https://microsoft.visualstudio.com/OS/_workitems/edit/53969605

### CODE REVIEW

File: onnxruntime/core/providers/openvino/openvino_provider_factory.cc [hunk 1]
```diff
@@ -77,7 +93,11 @@ struct OpenVINO_Provider : Provider {
   void* GetInfo() override { return &g_info; }
 
   std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory(const void* void_params) override {
-    auto& provider_options_map = *reinterpret_cast<const ProviderOptions*>(void_params);
+      typedef std::pair<const ProviderOptions*, const ConfigOptions&> buffer_t;
```
Comment (ankitm3k) [inline onnxruntime/core/providers/openvino/openvino_provider_factory.cc:96]:
```suggestion
      typedef std::pair<const ProviderOptions*, const ConfigOptions&> ConfigBuffer;
```

typedef naming case must be similar to the case of a class as the return type creates confusion whether the its a variable or a type while reading.

File: onnxruntime/core/session/provider_bridge_ort.cc [hunk 2]
```diff
@@ -1809,49 +1814,18 @@ ProviderOptions OrtOpenVINOProviderOptionsToOrtOpenVINOProviderOptionsV2(const O
 
   // Add new provider option below
   ov_options_converted_map["num_streams"] = "1";
-  ov_options_converted_map["export_ep_ctx_blob"] = "false";
+  //ov_options_converted_map["export_ep_ctx_blob"] = "false";
```
Comment (ankitm3k) [inline onnxruntime/core/session/provider_bridge_ort.cc:1817]: Please remove the redundant comment

File: onnxruntime/core/session/provider_bridge_ort.cc [hunk 3]
```diff
@@ -1809,49 +1814,18 @@ ProviderOptions OrtOpenVINOProviderOptionsToOrtOpenVINOProviderOptionsV2(const O
...
+std::shared_ptr<IExecutionProviderFactory> OpenVINOProviderFactoryCreator::Create(const ProviderOptions* provider_options_map,
                                                                                   const SessionOptions* session_options) {
   // Append session options applicable for EP to EP Provider options.
-  if (session_options) {
-    onnxruntime::ORTSessionOptionsToOrtOpenVINOProviderOptions(*provider_options_map, session_options);
-  }
-  return s_library_openvino.Get().CreateExecutionProviderFactory(provider_options_map);
+  std::pair<const ProviderOptions*, const ConfigOptions&> buffer = {provider_options_map, session_options->config_options};
+  const void* obj = reinterpret_cast<const void*>(&buffer);
+  return s_library_openvino.Get().CreateExecutionProviderFactory(obj);
```
Comment (ankitm3k) [inline onnxruntime/core/session/provider_bridge_ort.cc:1828]: Optional 
```suggestion
  return s_library_openvino.Get().CreateExecutionProviderFactory(reinterpret_cast<const void*>(&buffer));
```

File: onnxruntime/core/providers/openvino/openvino_provider_factory.cc (hunk 1 above)
Comment (ankitm3k) [reply onnxruntime/core/providers/openvino/openvino_provider_factory.cc:96]: Below can optimally look like -

const ConfigBuffer* buffer = reinterpret_cast<const ConfigBuffer*>(void_params);

**Comment by sfatimar** [comment]
LGTM

**Pull request merged**